### PR TITLE
Annotation module

### LINKED ISSUE #1: Implement Annotation module
Annotation module allows users to add annotation to their applications and allow us to record unique path for each trace. 

```
Path {
  path_id
 events: [Event]
}

Event = Task | Notice | Message

Task {
  name
  timestamp
  Task
}

Notice {
  name
  timestamp
}

Message {
  message_type
  message_id
  message_size
  timestamp
}

timestamp = process_name:path_id:vector_clock_value
message_type = send | receive

Functions:
- init(): initialize the state, create trace files. This should utilize Agent in elixir
- set_path_id: set path id for current path <- using random hash
- start_task, end_task: should be accompanied each other of a task
- notice(string): record a log message
- send(...)/receive(...): message_id and message_size must be the same for 1-1 send/receive messages
```

### PR DESCRIPTION
**Pull request by Kn99HN**
- Finish annotation module with basic test setup

### Note for reviewer
- If you can, do fetch and try to setup on your end to see if the test are running with `mix test` and setting up specified in the `README.md`. Traces should show up under `traces` directory.

### CODE REVIEW

**Comment by joyceluyy** [comment]
Hm I am getting following error. 

== Compilation error in file lib/path.ex ==
** (ArgumentError) Jason.Encoder is not available, cannot derive Jason.Encoder for Annotation.Path

**Comment by Kn99HN** [comment]
Can you try run `mix deps.get`?

File: apps/annotation/mix.exs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      extra_applications: [:logger]
+    ]
+  end
+
+  # Run "mix help deps" to learn about dependencies.
+  defp deps do
+    [
+      # {:dep_from_hexpm, "~> 0.3.0"},
+      # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"},
+      # {:sibling_app_in_umbrella, in_umbrella: true}
```
Comment (joyceluyy) [inline apps/annotation/mix.exs]: Do we need to add {:emulation, in_umbrella: true} here? I was running the code from a new vagrant

**Review by joyceluyy — COMMENTED** [review]
I added 
{:emulation, in_umbrella: true} then re-run the 'mix deps.get' and it is working

File: apps/annotation/mix.exs (hunk 1 above)
Comment (Kn99HN) [reply apps/annotation/mix.exs]: Ah you are right! Since we rely on Emulation in Annotation module, we need that line. Lemme add that real quick

**Review by joyceluyy — APPROVED** [review]

**Pull request merged**